wrapper/src/main/java/software/amazon/jdbc/util/ConnectionUrlParser.java

```diff
@@ -56,9 +59,9 @@ public List<HostSpec> getHostsFromConnectionUrl(final String initialConnection,
         final HostSpec host;
         if (singleWriterConnectionString) {
           final HostRole role = i > 0 ? HostRole.READER : HostRole.WRITER;
-          host = parseHostPortPair(hostArray[i], role);
+          host = parseHostPortPair(hostArray[i], role, new HostSpecBuilder(hostSpecBuilder));
```

Comment:
That's an interesting case. You need to use a host spec builder many times (since we're doing a loop) and you need a new builder. Can you use Supplier<HostSpecBuilder> as a parameter to this method?

Reply:
NOTE TO SELF: move `Supplier<HostSpecBuilder>` to method signature of `getHostsFromConnectionUrl`

Reply:
Done!